bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/ui/launch/JRETab.java

```diff
@@ -0,0 +1,74 @@
+package io.openliberty.tools.eclipse.ui.launch;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.jdt.debug.ui.launchConfigurations.JavaJRETab;
+import org.eclipse.jdt.launching.IVMInstall;
+import org.eclipse.jdt.launching.JavaRuntime;
+
+import io.openliberty.tools.eclipse.logging.Trace;
+import io.openliberty.tools.eclipse.utils.ErrorHandler;
+import io.openliberty.tools.eclipse.utils.Utils;
+
+public class JRETab extends JavaJRETab {
+
+    /** JRE container key. */
+    public static final String JRE_CONTAINER_KEY = "org.eclipse.jdt.launching.JRE_CONTAINER";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isValid(ILaunchConfiguration config) {
+        if (!super.isValid(config)) {
+            return false;
+        }
+
+        setWarningMessage(null);
+
+        // Issue a warning if we detect that the java installation is not a JDK.
+        String javaHome = resolveJavaHome(config);
+        java.nio.file.Path toolsJarPath = Paths.get(javaHome, "bin", (Utils.isWindows() ? "javac.exe" : "javac"));
+        File toolsJar = toolsJarPath.toFile();
+        if (!toolsJar.exists()) {
+            super.setWarningMessage("A Java Development Kit (JDK) is required to use Liberty dev mode.");
```

Comment:
I think we can go with the error message over the warning. Yes there's the risk we're wrong and need to ship a fix... but... if you run with the JRE I think it's not too obvious what the problem is.. it is maybe a bit hard to pick out in the Maven log.

I coded it up here:
https://github.com/OpenLiberty/liberty-tools-eclipse/commit/0dd62640fca237e70c9b7a8f03ebfa416d2eb4af